File: 03-3-PesquisaDeInteresseAppRefactored/app/src/main/java/com/example/pesquisadeinteresseapp/model/PreferenciaDAO.java
package com.example.pesquisadeinteresseapp.model;

import java.util.ArrayList;
import java.util.stream.Collectors;

public class PreferenciaDAO {
    ArrayList<Preferencia> listOfPrefs = new ArrayList<Preferencia>();

    public PreferenciaDAO() {
        //faixa 1
        Preferencia p = new Preferencia(1, "Hello Kitty", 1);
        listOfPrefs.add(p);
        p = new Preferencia(2, "Ben 10", 1);
        listOfPrefs.add(p);
        p = new Preferencia(3, "Rebeldes", 1);
        listOfPrefs.add(p);
        p = new Preferencia(4, "Chiquititas", 1);
        listOfPrefs.add(p);
        p = new Preferencia(5, "Transformers", 1);
        listOfPrefs.add(p);

        //faixa 2
        p = new Preferencia(6, "Game of Thrones", 2);
        listOfPrefs.add(p);
        p = new Preferencia(7, "Dragon Ball", 2);
        listOfPrefs.add(p);
        p = new Preferencia(8, "Jaspion", 2);
        listOfPrefs.add(p);

        //faixa 3
        p = new Preferencia(9, "Pantanal", 3);
        listOfPrefs.add(p);
        p = new Preferencia(10, "Faustao", 3);
        listOfPrefs.add(p);
        p = new Preferencia(11, "Zorra Total", 3);
        listOfPrefs.add(p);
        p = new Preferencia(12, "Ta no Ar", 3);
        listOfPrefs.add(p);
    }

    public ArrayList<Preferencia> getByFaixa(int faixa) {
        ArrayList<Preferencia> ret = new ArrayList<Preferencia>();
        for(Preferencia p : listOfPrefs)
            if(p.getFaixa() == faixa)
                ret.add(p);

        return ret;
    }
}

File: 03-3-PesquisaDeInteresseApp_kt2/app/src/main/java/com/example/pesquisadeinteresseapp_kt2/model/PreferenciaDAO.kt
package com.example.pesquisadeinteresseapp_kt2.model

class PreferenciaDAO {
    val listOfPrefs: MutableList<Preferencia> = mutableListOf()

    init {
        // faixa 1
        listOfPrefs.add(Preferencia(1, "Hello Kitty", 1))
        listOfPrefs.add(Preferencia(2, "Ben 10", 1))
        listOfPrefs.add(Preferencia(3, "Rebeldes", 1))
        listOfPrefs.add(Preferencia(4, "Chiquititas", 1))
        listOfPrefs.add(Preferencia(5, "Transformers", 1))

        // faixa 2
        listOfPrefs.add(Preferencia(6, "Game of Thrones", 2))
        listOfPrefs.add(Preferencia(7, "Dragon Ball", 2))
        listOfPrefs.add(Preferencia(8, "Jaspion", 2))

        // faixa 3
        listOfPrefs.add(Preferencia(9, "Pantanal", 3))
        listOfPrefs.add(Preferencia(10, "Faustao", 3))
        listOfPrefs.add(Preferencia(11, "Zorra Total", 3))
        listOfPrefs.add(Preferencia(12, "Ta no Ar", 3))
    }

    fun getByFaixa(faixa: Int): List<Preferencia> {
        val res = listOfPrefs.filter { it.faixaEtaria == faixa }
        return res.toList()
    }
}